src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelperBase.cs

```diff
@@ -201,6 +201,68 @@ protected string GenerateUrl(string protocol, string host, string virtualPath, s
             }
         }
 
+        /// <summary>
+        /// Generates a URI from the provided components.
+        /// </summary>
+        /// <param name="protocol">The URI scheme/protocol.</param>
+        /// <param name="host">The URI host.</param>
+        /// <param name="path">The URI path and remaining portions (path, query, and fragment).</param>
+        /// <returns>
+        /// An absolute URI if the <paramref name="protocol"/> or <paramref name="host"/> is specified, otherwise generates a
+        /// URI with an absolute path.
+        /// </returns>
+        protected string GenerateUrl(string protocol, string host, string path)
+        {
+            // This method is similar to GenerateUrl, but it's used for EndpointRouting. It ignores pathbase and fragment
+            // because those have already been incorporated.
```

Comment:
This is a copy paste of some existing code, with a few semantics changed and fewer parameters. I don't want to change the meaning of the existing code.